### PR TITLE
refact: merge pd into hugegraph

### PR DESCRIPTION
**Pull request by VGalaxies**
## Purpose of the PR

subtask of https://github.com/apache/incubator-hugegraph/issues/2265

Merge the internal version of pd into hugegraph.

## Main Changes

1. Merge the internal version of pd into `hugegraph-pd` module.
2. Modified some pom files under the `hugegraph-pd` module so that it can be built normally **under certain conditions**.
3. Clean some comments in `hugegraph-pd`.
4. Add `apache-rat-plugin` for hugegraph-pd and add some license headers.
5. **(TODO)** potential code cleanup or style issues

NOTE:

1. Due to the modification of `<parent>` in the root pom of `hugegraph-pd` from `spring-boot-starter-parent` to `hugegraph`, it is necessary to explicitly specify the version of some dependencies for the `hugegraph-pd` submodules.
2. The `hugegraph-pd` module incorporates features that are not supported in Java 8, such as the `var` keyword. To ensure compatibility with Java 8, subsequent PRs could consider adjusting the compilation version based on whether the user opts to use `hugegraph-pd` and `hugegraph-store`.

## Verifying these changes

- [ ] Trivial rework / code cleanup without any test coverage. (No Need)
- [ ] Already covered by existing tests, such as *(please modify tests here)*.
- [x] Need tests and can be verified as follows:
    - subsequent PRs will merge store and more adaptation work needs to be done

## Does this PR potentially affect the following parts?

- [ ]  Nope
- [ ]  Dependencies (add/update license info) 
- [ ]  Modify configurations
- [ ]  The public API
- [x]  Other affects (breaking changes)

## Documentation Status

- [x]  `Doc - TODO` 
- [ ]  `Doc - Done` 
- [ ]  `Doc - No Need`

### CODE REVIEW

File: hugegraph-pd/hg-pd-service/src/main/resources/private-keys.store
Comment (VGalaxies) [inline hugegraph-pd/hg-pd-service/src/main/resources/private-keys.store:1]: @imbajin Does this file need to be used?

File: hugegraph-pd/hg-pd-service/src/main/resources/public-certs.store
Comment (VGalaxies) [inline hugegraph-pd/hg-pd-service/src/main/resources/public-certs.store:1]: @imbajin ditto

File: hugegraph-pd/hg-pd-service/src/main/resources/private-keys.store
Comment (imbajin) [reply hugegraph-pd/hg-pd-service/src/main/resources/private-keys.store:1]: > @imbajin Does this file need to be used?

Nope, could check the reference & safe remove it

File: hugegraph-pd/hg-pd-service/src/main/resources/public-certs.store
Comment (imbajin) [reply hugegraph-pd/hg-pd-service/src/main/resources/public-certs.store:1]: > @imbajin ditto

same, we don't need license(related) files

**Review by imbajin — APPROVED** [review]
LGTM

**Pull request merged**